pandas/tests/indexes/test_engine.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import pytest
+
+import pandas as pd
+from pandas._libs.index import (Int64Engine, UInt64Engine,
+                                Float64Engine, ObjectEngine)
+
+
+class TestNumericEngine(object):
+
+    @pytest.mark.parametrize('data', [[0, 1, 2]])
+    def test_engine_type(self, data, num_engine):
+        index = pd.Index(data, dtype=num_engine._dtype)
+        if issubclass(index.dtype.type, np.signedinteger):
+            assert isinstance(index._engine, Int64Engine)
+        elif issubclass(index.dtype.type, np.unsignedinteger):
+            assert isinstance(index._engine, UInt64Engine)
+        elif issubclass(index.dtype.type, np.floating):
+            assert isinstance(index._engine, Float64Engine)
+        else:
+            raise TypeError("unexpected dtype {}".format(index.dtype))
+
+    @pytest.mark.parametrize('data', [[0, 1, 2]])
+    def test_is_monotonic_ordered(self, data, num_engine):
+        codes = np.array(data, dtype=num_engine._dtype)
```

Comment:
why do you have multiple tests for monotonic? and not monotonic?

Reply:
Yeah, there just one data to check for. Fixed.